### PR TITLE
Feature/cx 6785 x device scam warning customisation

### PR DESCRIPTION
**Pull request by kopijunkie**
# Feature

Allow integrators to be able to add their own branding text and/or logo image on the new Cross Device Client Intro screen in PR #1534 

cross-device-client-intro-example-2

- Integrator should be able to supply two optional configurations: 

  1. `crossDeviceIntroProductName` (string)
  2. `crossDeviceIntroProductLogoSrc` (string - URL)

- If `crossDeviceIntroProductName` is not specified, user will see the default subtitle copy of `"Continue with the verification"`. Otherwise, they will see the subtitle incorporating their product name, copy after the default copy.

- If `crossDeviceIntroProductLogo` is not specified, user will see the default icon. Otherwise, they will see their product logo.

- Added ARIA label for the integrator provided logo image of "Brand logo"

- Updated README with description of the new options and included logo dimension recommendation in the `README`,  an explanation of why they are recommended to set these options

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [x] Has the README been updated?
- [x] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 ## Added
 
 - UI: Added new Intro screen when user begins the Cross Device flow on their mobile device.
+- Public: Add SDK configuration options for integrators to add a custom product name, copy and/or custom product logo image in the new Cross Device mobile client Intro screen.
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply CHANGELOG.md:14]: Can confirm we should be using US English spelling after confirming with Ben I think sometime last year. But there's definitely some inconsistency in the existing docs. General rule of thumb I'm following now is to use US in the new content and update anything I spot but I admittedly have a tendency to auto type in UK English unless corrected during review or actively conscious of it myself.

File: README.md [hunk 2]
```diff
@@ -415,6 +417,36 @@ onfidoOut.tearDown()
   }
   ```
 
+- **`crossDeviceClientIntroProductName {String} optional`**
+
+  When a user switch to the SDK's Cross Device flow, they will see an intro screen when the SDK client loads on their mobile browser.
+  ![Default Cross Device Client Intro screen](demo/cross-device-client-intro.png)
+  You can customise the text by adding your company or product's name to the subtitle with this option. We recommend that you set this alongside the corresponding `crossDeviceClientIntroProductLogoSrc` below to make the user aware that this is part of a flow that was initiated on a desktop or laptop browser when they open the Cross Device link on their mobile browser. This is also an opportunity to include your branding in the SDK flow.
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply README.md:424]: US. Will correct it after committing your suggestion. Easier that way.

File: README.md [hunk 3]
```diff
@@ -415,6 +417,36 @@ onfidoOut.tearDown()
...
+  ```javascript
+  Onfido.init({
+    token: '<YOUR_SDK_TOKEN>',
+    crossDeviceClientIntroProductLogoSrc: 'path://to/logo/image/file',
+  })
+  ```
+
 - **`steps {List} optional`**
 
   The list of the different steps to be shown in the SDK flow and their custom options. Each step can either be specified as a string (when no customization is required) or an object (when customization is required).
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply README.md:452]: US spellin, so `customization` is correct.

File: README.md [hunk 4]
```diff
@@ -415,6 +417,36 @@ onfidoOut.tearDown()
   }
   ```
 
+- **`crossDeviceClientIntroProductName {String} optional`**
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply README.md:420]: Good suggestion. I didn't put it there as the documentation in "Flow customisation" section seemed to be all specific to SDK steps configurations and wanted to avoid any confusion about where to use it in the SDK configuration, but this could work.

File: README.md [hunk 5]
```diff
@@ -790,6 +792,38 @@ This is the final completion step. The screen displays a completion message to s
...
+    token: '<YOUR_SDK_TOKEN>',
+    crossDeviceClientIntroProductName: 'for a [COMPANY/PRODUCT NAME] loan',
+  })
+  ```
+
+  ![Cross Device Client Intro screen with client product name and copy](demo/cross-device-client-intro-example-1.png)
+
+- **`crossDeviceClientIntroProductLogoSrc {String} optional`**
+
+  You can customize the icon by adding your company or product's logo to be displayed instead of the default SDK icon image with this option. We recommend that you set this, alongside the corresponding `crossDeviceClientIntroProductName` above, to notify the user that this is part of a flow initiated on a desktop or laptop browser when they open the Cross Device link on their mobile browser. This is also an opportunity to include branding in the SDK flow.
```
Comment (Phoebe-B) [inline README.md:815]:
```suggestion
  You can customize the icon by adding your company or product logo to be displayed instead of the default SDK icon image with this option. We recommend that you set this, alongside the corresponding `crossDeviceClientIntroProductName` above, to notify the user that this is part of a flow initiated on a desktop or laptop browser when they open the Cross Device link on their mobile browser. This is also an opportunity to include your branding in the SDK flow.
```

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Phoebe-B) [reply CHANGELOG.md:14]: The changes to this sentence still need to be made/committed

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Phoebe-B) [reply README.md]: Looks good! Thanks

**Review by Phoebe-B — APPROVED** [review]

File: src/components/crossDevice/ClientIntro/index.tsx [hunk 6]
```diff
@@ -12,11 +12,21 @@ import style from './style.scss'
...
-const CrossDeviceClientIntro: FunctionComponent<Props> = ({ nextStep }) => {
+const CrossDeviceClientIntro: FunctionComponent<Props> = ({
+  nextStep,
+  crossDeviceClientIntroProductName,
+  crossDeviceClientIntroProductLogoSrc,
+}) => {
   const { translate } = useLocales()
+  const defaultSubtitle = translate('cross_device_session_linked.subtitle')
+  const subtitle = crossDeviceClientIntroProductName
+    ? `${defaultSubtitle} ${crossDeviceClientIntroProductName}`
```
Comment (ethan605) [inline src/components/crossDevice/ClientIntro/index.tsx:28]: String template in this situation is a bit hard to follow. I'd suggest something more traditional:
```suggestion
    ? `[defaultSubtitle, crossDeviceClientIntroProductName].join(' ')`
```

File: src/types/sdk.ts [hunk 7]
```diff
@@ -68,6 +73,9 @@ export interface SdkOptions extends FunctionalConfigurations {
   steps?: Array<StepTypes | StepConfig>
   enterpriseFeatures?: EnterpriseFeatures
   customUI?: UICustomizationOptions | null
+  crossDeviceClientIntro?: CrossDeviceClientIntroOptions
```
Comment (ethan605) [inline src/types/sdk.ts:76]: Are we accepting both flattened and nested options for this feature? 🤔

**Review by ethan605 — CHANGES_REQUESTED** [review]
Just a minor suggestion & a clarification question

File: src/types/sdk.ts (hunk 7 above)
Comment (kopijunkie) [reply src/types/sdk.ts:76]: No, we're not. I was initially considering implementing it as a nested option but decided against it. Missed this when cleaning it up.

File: src/demo/SidebarSections.tsx [hunk 8]
```diff
@@ -134,8 +128,37 @@ export const SdkOptionsView: FunctionComponent<{
...
       </div>
-    </div>
-
+    </label>
+    <label>
+      crossDeviceClientIntroProductName
+      <div>
+        <input
+          type="text"
+          value={sdkOptions.crossDeviceClientIntroProductName || ''}
```
Comment (stefaniacardenas) [inline src/demo/SidebarSections.tsx:137]: What if the string passed is `"/><script>alert('xss')</script>""`?

File: src/demo/demoUtils.ts [hunk 9]
```diff
@@ -295,6 +298,15 @@ export const getInitSdkOptions = (): SdkOptions => {
   const customUI =
     queryParamToValueString.customisedUI === 'true' ? customUIConfig : undefined
 
+  const crossDeviceClientIntroProductName =
+    queryParamToValueString.crossDeviceClientIntroCustomProductName === 'true'
```
Comment (stefaniacardenas) [inline src/demo/demoUtils.ts:302]: In the contributor's file it says that this query string is expecting a string, not a boolean

File: src/components/crossDevice/ClientIntro/index.tsx (hunk 6 above)
Comment (stefaniacardenas) [reply src/components/crossDevice/ClientIntro/index.tsx]: No need for `` in the above suggestion

**Review by stefaniacardenas — CHANGES_REQUESTED** [review]

File: src/components/crossDevice/ClientIntro/index.tsx [hunk 10]
```diff
@@ -12,11 +12,21 @@ import style from './style.scss'
...
-const CrossDeviceClientIntro: FunctionComponent<Props> = ({ nextStep }) => {
+const CrossDeviceClientIntro: FunctionComponent<Props> = ({
+  nextStep,
+  crossDeviceClientIntroProductName,
+  crossDeviceClientIntroProductLogoSrc,
+}) => {
   const { translate } = useLocales()
+  const defaultSubtitle = translate('cross_device_session_linked.subtitle')
+  const subtitle = crossDeviceClientIntroProductName
+    ? `[defaultSubtitle, crossDeviceClientIntroProductName].join(' ')`
```
Comment (stefaniacardenas) [inline src/components/crossDevice/ClientIntro/index.tsx:28]:
```suggestion
    ? [defaultSubtitle, crossDeviceClientIntroProductName].join(' ')
```

**Review by stefaniacardenas — CHANGES_REQUESTED** [review]

File: src/demo/SidebarSections.tsx (hunk 8 above)
Comment (kopijunkie) [reply src/demo/SidebarSections.tsx:137]: It will get displayed as a plain string and the alert doesn't get executed since these are in JSX.

File: src/demo/demoUtils.ts (hunk 9 above)
Comment (kopijunkie) [reply src/demo/demoUtils.ts:302]: Good catch! Confused it with the integration documentation in the contributing file.

File: src/components/crossDevice/ClientIntro/index.tsx (hunk 10 above)
Comment (kopijunkie) [reply src/components/crossDevice/ClientIntro/index.tsx:28]: Missed this when I committed the suggestion. Thanks!

**Review by stefaniacardenas — APPROVED** [review]

**Review by ethan605 — APPROVED** [review]
